jOOQ/src/main/java/org/jooq/GrantGrantedStep.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Other licenses:
+ * -----------------------------------------------------------------------------
+ * Commercial licenses for this work are available. These replace the above
+ * ASL 2.0 and offer limited warranties, support, maintenance, and commercial
+ * database integrations.
+ *
+ * For more information, please visit: http://www.jooq.org/licenses
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ */
+package org.jooq;
+
+import static org.jooq.SQLDialect.FIREBIRD;
+import static org.jooq.SQLDialect.HSQLDB;
+import static org.jooq.SQLDialect.MYSQL;
+import static org.jooq.SQLDialect.POSTGRES;
+
+/**
+ * The step of creation <code>GRANT</code> statement that assign user
+ * or role a grant option.
+ *
+ * @author Timur Shaidullin
+ */
+public interface GrantGrantedStep extends GrantFinalStep{
```

Comment:
Usually, the naming scheme is indicating what's possible in the current step, not what has been done in the previous step(s) as that could be ambiguous. So, a good name would be `GrantWithGrantOptionStep` or `GrantGrantOptionStep`. I'll fix this after merging.

Reply:
Name `GrantGrantOptionStep` is more suitable for this case, it lets to add `[ GRANTED BY <grantor> ]` in the future.

Reply:
Yeah, perhaps, although I'd like to wait for someone to actually request the `GRANTED BY <grantor>` feature before implementing that...